libass/ass_fontselect.c

```diff
@@ -168,22 +168,34 @@ static void load_fonts_from_dir(ASS_Library *library, const char *dir)
     DIR *d = opendir(dir);
     if (!d)
         return;
+    size_t dirlen = strlen(dir);
+    size_t namelenmax = FFMAX(2048, dirlen + 256);
+    char *namebuf = malloc(namelenmax);
+    if (!namebuf)
+        return;
     while (1) {
         struct dirent *entry = readdir(d);
         if (!entry)
             break;
         if (entry->d_name[0] == '.')
             continue;
-        char fullname[4096];
-        snprintf(fullname, sizeof(fullname), "%s/%s", dir, entry->d_name);
+        size_t namelen = dirlen + strlen(entry->d_name) + 2;
+        if (namelen > namelenmax) {
+            if (ASS_REALLOC_ARRAY(namebuf, namelen))
```

Comment:
There is not much sense to limit the buffer to currently required size only. If there is one file with a long name then probability of existence of another file with similarly long name is significantly increased. So I think it's better to enlarge `namelen` prior to realloc.